backend/inspirehep/serializers.py

```diff
@@ -321,3 +322,76 @@ def dumps(obj, app=None, **kwargs):
             return rv.encode(encoding)
 
     return rv
+
+
+def record_responsify(serializer, mimetype):
+    """Create a Records-REST response serializer.
+    :param serializer: Serializer instance.
+    :param mimetype: MIME type of response.
+    :returns: Function that generates a record HTTP response.
+    """
+
+    def view(pid, record, code=200, headers=None, links_factory=None):
+        response = current_app.response_class(
+            serializer.serialize(pid, record, links_factory=links_factory),
+            mimetype=mimetype,
+        )
+        response.status_code = code
+        set_headers_for_record_caching_and_concurrency(response, record)
+
+        if headers is not None:
+            response.headers.extend(headers)
+
+        if links_factory:
+            add_link_header(response, links_factory(pid))
+
+        return response
+
+    return view
+
+
+def search_responsify(serializer, mimetype):
+    """Create a Records-REST search result response serializer.
+    :param serializer: Serializer instance.
+    :param mimetype: MIME type of response.
+    :returns: Function that generates a record HTTP response.
+    """
+
+    def view(
+        pid_fetcher,
+        search_result,
+        code=200,
+        headers=None,
+        links=None,
+        item_links_factory=None,
+    ):
+        response = current_app.response_class(
+            serializer.serialize_search(
+                pid_fetcher,
+                search_result,
+                links=links,
+                item_links_factory=item_links_factory,
+            ),
+            mimetype=mimetype,
+        )
+        response.status_code = code
+        if headers is not None:
+            response.headers.extend(headers)
+
+        if links:
```

Comment:
why we have different for search and record? shouldn't be something like `links_to_header True or False`